libraries/BufferedBleSpp/Xamarin Forms/BufferedBleSpp.cs

```diff
@@ -0,0 +1,153 @@
+﻿using Plugin.BLE.Abstractions.Contracts;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace BufferedBleSppXamarin
+{
+    internal class BufferedBleSpp
+    {
+        // RX... and TX... naming is from server viewpoint, not the client (this code)
+
+        const int BUFFER_SIZE = 250;
+
+        private Guid ServiceUUID = new Guid("12345678-1234-5678-1234-56789abcdef0");

```

Comment:
private readonly for all those as they are not changing and you don't want them to change obviously :-)

Reply:
Also, in .NET nanoFrameork, we are using underscore `_` to start any variable. It makes it easier in a function to understand if it's a global variable or a local one. It does avoid a lot of confusion and avoid usage of this.yourvariable = yourvariable for example